src/components/icons.js

```diff
@@ -14,7 +14,16 @@ export const icon = (shape, { size = 16, ...props } = {}) => _.invokeArgs(shape,
 
 export const spinner = props => icon('loadingSpinner', _.merge({ size: 24, style: { color: colors.primary() } }, props))
 
-export const centeredSpinner = props => spinner(_.merge({ size: 48, style: { display: 'block', margin: 'auto' } }, props))
+export const centeredSpinner = ({ size = 48, ...props } = {}) => spinner(_.merge({
+  size, style: {
+    display: 'block',
+    position: 'sticky',
+    top: `calc(50% - ${size / 2}px)`,
```

Comment:
This trick using css `calc` is neat, I looked it up to be sure I fully understood what it was doing and this certainly seems like an appropriate place for it, nice!